packages/flutter/lib/src/cupertino/scaffold.dart

```diff
@@ -0,0 +1,157 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+import 'bottom_tab_bar.dart';
+import 'nav_bar.dart';
+
+/// Implements the basic iOS application's layout and behavior structure.
+///
+/// The scaffold lays out the navigation bar on top, the tab bar at the bottom
+/// and tabbed or untabbed content between or behind the bars.
+///
+/// For tabbed scaffolds, the tab's active item and the actively showing tab
+/// in the content area are automatically connected.
+// TODO(xster): describe navigator handlings.
+// TODO(xster): add an example.
+class CupertinoScaffold extends StatefulWidget {
+  /// Construct a [CupertinoScaffold] without tabs.
+  // TODO(xster): document that page transitions will happen behind the navigation
+  // bar.
+  const CupertinoScaffold({
+    Key key,
+    this.navigationBar,
+    @required this.child,
+  }) : tabBar = null,
+       rootPageBuilder = null,
+       super(key: key);
+
+  /// Construct a [CupertinoScaffold] with tabs. A [tabBar] and a [rootPageBuilder]
```

Comment:
Usually you want the first paragraph to be just one sentence. The first paragraph is what ends up being used as the "short doc" in the API doc summaries.